packages/gitgraph-react/src/Gitgraph.tsx

```diff
@@ -227,19 +196,32 @@ class Gitgraph extends React.Component<GitgraphProps, GitgraphState> {
     }
 
     return (
-      <g
-        key={commit.hashAbbrev}
-        // Store commit ID to match branch label. We could use refs instead.
-        id={commit.hashAbbrev}
-        transform={`translate(${x}, ${y})`}
-      >
+      <g key={commit.hashAbbrev} transform={`translate(${x}, ${y})`}>
         {this.renderDot(commit)}
-        {commit.style.message.display && this.renderMessage(commit)}
         {this.gitgraph.template.arrow.size && this.renderArrows(commit)}
+
+        {/* These elements are positionned after component update. */}
+        <g transform={`translate(${-x}, 0)`}>
```

Comment:
I'm considering to refactor the code by extracting these "commit elements" into their own component.